include/boost/geometry/strategies/spherical/line_interpolate_point.hpp

```diff
@@ -0,0 +1,175 @@
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_STRATEGIES_SPHERICAL_LINE_INTERPOLATE_POINT_HPP
+#define BOOST_GEOMETRY_STRATEGIES_SPHERICAL_LINE_INTERPOLATE_POINT_HPP
+
+#include <boost/geometry/core/assert.hpp>
+#include <boost/geometry/core/coordinate_dimension.hpp>
+#include <boost/geometry/core/coordinate_type.hpp>
+#include <boost/geometry/core/radian_access.hpp>
+#include <boost/geometry/srs/spheroid.hpp>
+#include <boost/geometry/strategies/line_interpolate_point.hpp>
+#include <boost/geometry/strategies/spherical/distance_haversine.hpp>
+#include <boost/geometry/util/select_most_precise.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace line_interpolate_point
+{
+
+
+/*!
+\brief Interpolate point on a spherical segment.
+\ingroup strategies
+\tparam CalculationType \tparam_calculation
+\tparam DistanceStrategy The underlying point-point distance strategy
+
+\qbk{
+[heading See also]
+\* [link geometry.reference.algorithms.line_interpolate_point.line_interpolate_point_4_with_strategy line_interpolate_point (with strategy)]
+}
+
+ */
+template
+<
+    typename CalculationType = void,
+    typename DistanceStrategy = distance::haversine<double, CalculationType>
+>
+class spherical
+{
+public:
+
+    //result type
+    template <typename Point>
+    struct result_type
+    {
+        typedef typename select_most_precise
+            <
+                typename coordinate_type<Point>::type,
+                CalculationType
+            >::type calc_t;
+
+        result_type(calc_t d) :
+            distance(d)
+        {}
+
+        calc_t distance;
+    };
+
+    template <typename Point>
+    inline result_type<Point> compute(Point const& p0,
+                                      Point const& p1) const
+    {        
+        return result_type<Point>(DistanceStrategy().apply(p0,p1));
+    }   
+
+    template <typename Point, typename T1, typename T2>
+    inline void apply(Point const& p0,
+                      Point const& p1,
+                      T1 const& fraction,
+                      Point & p,
+                      T2 const& dist_az) const
+    {
+        typedef typename select_most_precise
+            <
+                typename coordinate_type<Point>::type,
+                CalculationType
+            >::type calc_t;
+
+        calc_t const c0 = 0;
+        calc_t const c1 = 1;
+        calc_t const pi = math::pi<calc_t>();
+
+        typedef model::point<calc_t, 3, cs::cartesian> point3d_t;
+        point3d_t const xyz0 = formula::sph_to_cart3d<point3d_t>(p0);
+        point3d_t const xyz1 = formula::sph_to_cart3d<point3d_t>(p1);
+        //calc_t const dot01 = geometry::dot_product(xyz0, xyz1);
+        //calc_t const angle01 = acos(dot01);
+        calc_t const angle01 = dist_az.distance;
+
+        point3d_t axis;
```

Comment:
This code is from `densify` strategy, isn't it?
From here: https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/strategies/spherical/densify.hpp#L99
We should probably move it to formulas.